File: src/compiler.py
#Author: Judah Munoz

class Vector:
    def __init__(self, x,y,z):
        self.x = x
        self.y = y
        self.z = z

def stoVector(s):
    split = s.split()
    return Vector(split[1], split[2], split[3])

def vectortoString(v):
    return v.x + "," + v.y + "," + v.z

def peek_line(f):
    pos = f.tell()
    line = f.readline()
    f.seek(pos)
    return line

def compile_vertices(infile):
    line = "null";
    vectorList = [];

    #Burn through header
    while line[0] != 'v':
        line = infile.readline()
    
    #Add first vector to list
    vectorList.append(stoVector(line))

    #Keep adding while file contains vectors
    while peek_line(infile)[0] == 'v':
        line = infile.readline()
        vectorList.append(stoVector(line))

    return vectorList

if __name__ == "__main__":
    infile = open("input/line_cube.obj")
    
    verticeList = compile_vertices(infile)

    for v in verticeList:
        print(vectortoString(v) + "\n")

File: src/compiler.cpp
//Author: Judah Munoz

#include <iostream>
#include <fstream>

#include "constants.h"
#include "commands.h"
#include "vector.h"

using namespace std;

void setup(ofstream outfile){
    //Home axes
    outfile << MOVE_ORIGIN << "\n";
    //Reset extruder
    outfile << RESET_EXTRUDER << "\n";
    //Set speed
    outfile << SET_SPEED(SPEED) << "\n";
    //Set units to MM
    outfile << SET_MM << "\n";
};

void compile_vertices(ifstream infile){
    string line = "n";
    char peek;
    Vector vn;

    //Ignore header input
    while (line.substr(0,1) != "v"){
        getline(infile, line);
    }

    //Read first vector
    vn = stoVector(line.substr(2,line.length()), " ");
    peek = infile.peek();
    //Add to array

    //Read all the vertices
    while (peek == 'v'){
        getline(infile, line);
        vn = stoVector(line.substr(2,line.length()), " ");
        //Add to array
    }
}